src/analyzer/protocol/asn1/asn1.pac

```diff
@@ -183,17 +183,13 @@ zeek::StringValPtr asn1_oid_to_val(const ASN1Encoding* oid)
 
 		if ( i > 0 )
 			{
-			rval += ".";
-			snprintf(tmp, sizeof(tmp), "%" PRIu64, subidentifier_values[i]);
+			snprintf(tmp, sizeof(tmp), ".%" PRIu64, subidentifier_values[i]);
 			rval += tmp;
 			}
 		else
 			{
 			std::div_t result = std::div(subidentifier_values[i], 40);
-			snprintf(tmp, sizeof(tmp), "%d", result.quot);
-			rval += tmp;
-			rval += ".";
-			snprintf(tmp, sizeof(tmp), "%d", result.rem);
+			snprintf(tmp, sizeof(tmp), "%d.%d", result.quot, result.rem);
```

Comment:
Since this is now writing both numbers into `tmp` can this fail to format correctly if it exceeds the space in

```char tmp[32];``` ?  Not sure what the range of those values might be.

Reply:
`std::div` returns a structure with two `int`s in it. 32 characters should be enough to hold the range of both of them concatenated.